src/App.tsx

```diff
@@ -15,36 +17,92 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortType {
+  length = 'length',
+  alph = 'alph',
+}
+
+export const App = () => {
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortType, setSortType] = useState('');
+
+  const reverseList = () => {
+    if (isReversed) {
+      setIsReversed(false);
+    } else {
+      setIsReversed(true);
+    }
+  };
+
+  const resetList = () => {
+    setIsReversed(false);
+    setSortType('');
+  };
+
+  const prepearedListOfGoods = [...goodsFromServer];
+
+  if (sortType) {
+    prepearedListOfGoods.sort((good1, good2) => {
+      switch (sortType) {
+        case SortType.length:
+          return good1.length - good2.length;
+        case SortType.alph:
+          return good1.localeCompare(good2);
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    prepearedListOfGoods.reverse();
+  }
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={`button is-info ${cn({ 'is-light': sortType !== 'alph' })}`}
+          onClick={() => setSortType('alph')}
```

Comment:
use enum values instead of string literals here and anywhere you use _"setSortType(SortType.value)"_